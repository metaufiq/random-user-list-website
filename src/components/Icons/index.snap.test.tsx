import { SearchIcon, SortIcon, SortIconAsc, SortIconDesc } from ".";
import { assertSnapshots } from "../../../test";
import { Props } from "./index.types";

describe('SortIconSnapshot', () => {
  const sizeOnlyProps:Props = {
    size: 2
  };

  const props:Props = {
    size: 2,
    color: 'red'
  }

  const configs = [
    {
      props,
      desc: 'should render normal icon'
    },
    {
      props: sizeOnlyProps,
      desc: 'should render icon with custom color'
    },

  ];

  assertSnapshots(SortIcon, configs);
});

describe('SortIconAscSnapshot', () => {
  const sizeOnlyProps:Props = {
    size: 2
  };

  const props:Props = {
    size: 2,
    color: 'red'
  }

  const configs = [
    {
      props,
      desc: 'should render normal icon'
    },
    {
      props: sizeOnlyProps,
      desc: 'should render icon with custom color'
    },

  ];

  assertSnapshots(SortIconAsc, configs);
});

describe('SortIconDescSnapshot', () => {
  const sizeOnlyProps:Props = {
    size: 2
  };

  const props:Props = {
    size: 2,
    color: 'red'
  }

  const configs = [
    {
      props,
      desc: 'should render normal icon'
    },
    {
      props: sizeOnlyProps,
      desc: 'should render icon with custom color'
    },

  ];

  assertSnapshots(SortIconDesc, configs);
});

describe('SearchIconSnapshot', () => {
  const sizeOnlyProps:Props = {
    size: 2
  };

  const props:Props = {
    size: 2,
    color: 'red'
  }

  const configs = [
    {
      props,
      desc: 'should render normal icon'
    },
    {
      props: sizeOnlyProps,
      desc: 'should render icon with custom color'
    },

  ];

  assertSnapshots(SearchIcon, configs);
});