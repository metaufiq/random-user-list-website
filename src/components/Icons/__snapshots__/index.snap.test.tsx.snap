// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortIconAscSnapshot should render icon with custom color 1`] = `
<svg
  height={2}
  viewBox="0 0 24 24"
  width={2}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 0l-8 10h16l-8-10zm3.839 16l-3.839 4.798-3.839-4.798h7.678zm4.161-2h-16l8 10 8-10z"
  />
</svg>
`;

exports[`SortIconAscSnapshot should render normal icon 1`] = `
<svg
  height={2}
  viewBox="0 0 24 24"
  width={2}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 0l-8 10h16l-8-10zm3.839 16l-3.839 4.798-3.839-4.798h7.678zm4.161-2h-16l8 10 8-10z"
  />
</svg>
`;

exports[`SortIconDescSnapshot should render icon with custom color 1`] = `
<svg
  height={2}
  viewBox="0 0 24 24"
  width={2}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 3.202l3.839 4.798h-7.678l3.839-4.798zm0-3.202l-8 10h16l-8-10zm8 14h-16l8 10 8-10z"
  />
</svg>
`;

exports[`SortIconDescSnapshot should render normal icon 1`] = `
<svg
  height={2}
  viewBox="0 0 24 24"
  width={2}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 3.202l3.839 4.798h-7.678l3.839-4.798zm0-3.202l-8 10h16l-8-10zm8 14h-16l8 10 8-10z"
  />
</svg>
`;

exports[`SortIconSnapshot should render icon with custom color 1`] = `
<svg
  height={2}
  viewBox="0 0 24 24"
  width={2}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 0l8 10h-16l8-10zm8 14h-16l8 10 8-10z"
  />
</svg>
`;

exports[`SortIconSnapshot should render normal icon 1`] = `
<svg
  height={2}
  viewBox="0 0 24 24"
  width={2}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 0l8 10h-16l8-10zm8 14h-16l8 10 8-10z"
  />
</svg>
`;
